test/routes/home_route_test.dart

```diff
@@ -141,6 +148,9 @@ void main() {
         tester.getRect(find.byType(App)).height,
         reason: 'Player route must be offscreen',
       );
+      await tester.runAsync(() => tester.pump()); // Wait for widget events from start to process.
```

Comment:
This comment is even more confusing than the one above

Reply:
I've reworded it. This is another workaround to let some of the callback on streams called in `ContentControl.init` fire, so the widget events reach the app method channel.

I think it's this chain that needs to complete: `ContentControl.init` -> `MusicPlayer.instance.init` -> `restoreLastSong` -> `setSong` -> `PlaybackControl.instance.changeSong` -> `PlaybackControl.onSongChange.add` -> triggers `AppWidgetControl.update`.